src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -55,9 +55,10 @@ private static IResource PromoteAzureResourceFromAnnotation(IResource resource)
 
     public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
     {
-        // TODO: Make this more general purpose
         if (executionContext.IsPublishMode)
         {
+            // TODO: Let the user pick what this is, for now just hard code azure container apps
+            await new AzureContainerAppsInfastructure(executionContext).GenerateAdditionalInfrastructureAsync(appModel, cancellationToken).ConfigureAwait(false);
```

Comment:
I'm thinking that we'll want a model where we can choose per app/resource. This is fine for a x.1 release though. Just pointing to the horizon from here.

Reply:
We'll need to enable that for functions, for sure, but that can't be the default mode.